__tests__/components/TransactionList/TransactionRow.spec.ts

```diff
@@ -0,0 +1,888 @@
+                { idHex: '119C8A107ACADD5F', amount: 100 },
+                { idHex: '2C56C7D764F17B09', amount: 100 },
+                { idHex: '5B2DCC92C3B0FFD8', amount: 100 },
+                { idHex: '4A1D5DA123186518', amount: 100 },
+            ];
+
+            const mosaics = mockMosaic.map((mosaic) => new Mosaic(new MosaicId(mosaic.idHex), UInt64.fromUint(mosaic.amount)));
+
+            const wrapper = getTransactionRowWrapper(
+                { currentProfile: testnetProfile, networkMosaicId: networkMock.currency.mosaicIdHex, balanceMosaics: mosaicsMock },
+                { transaction: mockTransferTransaction(mosaics) },
+            );
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // Act:
+            const mosaicList = vm.nonNativeMosaicList();
+
+            // Act + Assert:
+            expect(mosaicList).toEqual([
+                { name: '461A9811A1DC1FBB', relativeAmount: '100' },
+                { name: '119C8A107ACADD5F', relativeAmount: '0.1' },
+                { name: '2C56C7D764F17B09', relativeAmount: '100' },
+                { name: '5B2DCC92C3B0FFD8', relativeAmount: '100' },
+                { name: '4A1D5DA123186518', relativeAmount: '100' },
+            ]);
+            expect(mosaicList.length).toBe(5);
+        });
+
+        test('returns empty non network mosaic when transaction type is non transfer transaction', () => {
+            // Arrange:
+            const wrapper = getTransactionRowWrapper(
+                { currentProfile: testnetProfile },
+                { transaction: mockAccountAddressRestrictionTransaction() },
+            );
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // Act:
+            const mosaicList = vm.nonNativeMosaicList();
+
+            // Act + Assert:
+            expect(mosaicList).toEqual([]);
+        });
+    });
+
+    describe('needsCosignature', () => {
+        const runBasicNeedsCosignatureTests = (transaction, expectedResult) => {
+            // Arrange:
+            const multisigAccounts = new Map<number, MultisigAccountInfo[]>();
+
+            const wrapper = getTransactionRowWrapper(
+                {
+                    currentProfile: testnetProfile,
+                    currentAccount: WalletsModel1,
+                    multisigAccountGraph: (multisigAccounts as unknown) as MultisigAccountInfo[][],
+                },
+                { transaction },
+            );
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // @ts-ignore - for private method
+            vm.aggregateTransactionDetails = transaction;
+
+            // Act:
+            vm.needsCosignature();
+
+            // Assert:
+            // @ts-ignore - for private method
+            expect(vm.transactionSigningFlag).toBe(expectedResult);
+        };
+
+        test('set transactionSigningFlag to false when current account is multisig', () => {
```

Comment:
do you mean aggregate complete? or why is flag not set ?

Reply:
`transactionSigningFlag` is used for cosigning transactions.
If the current account is multisig account, it should not have any consigning interface pop up.